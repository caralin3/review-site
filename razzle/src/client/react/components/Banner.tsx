import React from 'react';

export interface BannerProps {}

export const Banner: React.FC<BannerProps> = ({ children }) => (
  <div className="banner">{children}</div>
);
