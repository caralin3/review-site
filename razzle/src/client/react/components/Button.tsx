import React from 'react';

export interface ButtonProps {
  onClick: () => void;
  size?: 'sm' | 'md' | 'lg';
  text: string;
  variant: 'primary' | 'secondary';
}

export const Button: React.FC<ButtonProps> = ({
  onClick,
  size,
  text,
  variant
}) => (
  <button className={`btn-${variant}`} onClick={onClick}>
    {text}
  </button>
);
