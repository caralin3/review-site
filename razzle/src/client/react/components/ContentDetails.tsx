import React from 'react';
import { Link } from 'react-router-dom';
import { Content, User } from '../../../common';
import { routes } from '../routes';
import { StarRating, WatchButton } from '.';
import { user1 } from '../../mock';

export interface ContentDetailsProps {
  content: Content;
  onRate: (star: number) => void;
  onWatch: () => void;
  user?: User;
}

export const ContentDetails: React.FC<ContentDetailsProps> = ({
  content,
  onRate,
  onWatch,
  user = user1
}) => {
  const titleSection = (
    <div className="content__header">
      <span className="content__title">
        <div>{content.title}</div>
        {content.network && (
          <span className="content__network">{content.network}</span>
        )}
      </span>
      <StarRating
        rating={content.rating}
        myRating={content.myRating}
        onClick={onRate}
      />
    </div>
  );

  const detailsSection = (
    <div className="content__details">
      <p>{content.type}</p>
      <p>
        {content.type === 'Movie'
          ? content.year
          : `${content.year}-${content.endYear ? content.endYear : ''}`}
      </p>
      <p>{content.mpa}</p>
      <p>{content.duration} mins</p>
      {content.genres && (
        <p className="show-medium">{content.genres.toString()}</p>
      )}
      {content.genres && (
        <p className="show-small">{content.genres.slice(0, 2).toString()}</p>
      )}
    </div>
  );

  const peopleSection = (
    <>
      {content.director && (
        <p>
          <strong>
            {content.type === 'Movie' ? 'Director' : 'Creator'}:&nbsp;&nbsp;
          </strong>
          {content.director}
        </p>
      )}
      {content.actors && (
        <p>
          <strong>Actors:&nbsp;&nbsp;</strong>
          {content.actors.toString()}
        </p>
      )}
    </>
  );

  return (
    <div className="content">
      <div className="show-small">
        {titleSection}
        {detailsSection}
      </div>
      <div className="content__mobile show-small">
        <img
          className="content__image"
          src={content.image}
          alt="poster image"
        />
        <p className="content__synopsis">
          {content.synopsis && content.synopsis.substring(0, 175)}
          {content.synopsis && content.synopsis.length > 175 && '...'}
          <span className="content__footer-mobile">
            {user && (
              <WatchButton watching={content.watchList} onClick={onWatch} />
            )}
          </span>
        </p>
      </div>
      <div className="show-small">{peopleSection}</div>
      <img
        className="content__image show-medium"
        src={content.image}
        alt="poster image"
      />
      <div className="content__text show-medium">
        <div>
          {titleSection}
          {detailsSection}
        </div>
        <p className="content__synopsis">
          {content.synopsis && content.synopsis.substring(0, 450)}
          {content.synopsis && content.synopsis.length > 450 && '...'}
        </p>
        {peopleSection}
        <p className="content__footer">
          {user && (
            <WatchButton watching={content.watchList} onClick={onWatch} />
          )}
        </p>
      </div>
    </div>
  );
};
