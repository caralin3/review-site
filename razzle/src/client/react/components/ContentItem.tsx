import React from 'react';
import {
  Content,
  MultipleEpisodesResponse,
  MultipleReviewsResponse,
  Review as ReviewType,
  NewReview,
  NewReviewRequest,
  User
} from '../../../common';
import {
  ContentDetails,
  EpisodeItem,
  Review,
  ReviewEditor
} from '../components';
import { Link } from 'react-router-dom';
import { routes } from '../routes';

export interface ContentItemProps {
  content: Content;
  episodes?: MultipleEpisodesResponse;
  handleRating: (value: number, id: string, rated: boolean) => void;
  handleCreateReview: (body: NewReviewRequest) => void;
  handleDeleteReview: (id: string) => void;
  handleWatch: (watching: boolean, id: string) => void;
  reviews?: MultipleReviewsResponse;
  reviewsError?: Error;
  reviewsLoading: boolean;
  user?: User;
}

export const ContentItem: React.FC<ContentItemProps> = ({
  content,
  episodes,
  handleCreateReview,
  handleDeleteReview,
  handleRating,
  handleWatch,
  reviews,
  reviewsError,
  reviewsLoading,
  user
}) => {
  const [review, setReview] = React.useState<NewReview>({} as NewReview);
  const [errors, setErrors] = React.useState<string[]>([]);
  const [loading, setLoading] = React.useState(false);
  const [submit, setSubmit] = React.useState(false);

  const reviewsList: ReviewType[] =
    reviews && reviews.reviews ? reviews.reviews : [];

  const reviewed = reviewsList.filter(
    rev => user && rev.author.username === user.username
  );

  const handleSubmit = async () => {
    setSubmit(true);
    if (!review.rating) {
      setErrors(['Rating required']);
      return;
    }
    if (review.body && review.rating) {
      setLoading(true);
      try {
        await handleCreateReview({ review });
        setSubmit(false);
      } catch (err) {
        console.error(err);
        if (err.response) {
          const apiErrors = err.response.data.errors;
          const formErrors: string[] = [];
          Object.keys(apiErrors).forEach(key => {
            if (apiErrors) {
              formErrors.push(`${key} ${apiErrors[key]}`);
            }
          });
          setErrors(formErrors);
        }
      }
    }
    setLoading(false);
  };

  const resetForm = () => {
    setErrors([]);
    setLoading(false);
    setSubmit(false);
  };

  return (
    <div className="content-page">
      <section>
        <ContentDetails
          content={content}
          onRate={(val, rated) => handleRating(val, content.id, rated)}
          onWatch={() => handleWatch(content.watchList, content.id)}
          user={user}
        />
      </section>
      {episodes && (
        <section>
          <h2 className="content-page__header">Episodes</h2>
          <ul className="episode-item__list">
            {episodes.episodes.length > 0 &&
              episodes.episodes.map(ep => (
                <li className="episode-item__list-item" key={ep.id}>
                  <EpisodeItem episode={ep} />
                </li>
              ))}
          </ul>
        </section>
      )}
      <section>
        <h2 className="content-page__header">Reviews</h2>
        <div className="content-page__reviews">
          {user && reviewed.length === 0 && (
            <ReviewEditor
              errors={errors}
              loading={loading}
              onChange={e => {
                setReview({ ...review, body: e.target.value });
                resetForm();
              }}
              onRate={val => {
                setReview({ ...review, rating: val });
                resetForm();
              }}
              onSubmit={handleSubmit}
              rating={review.rating}
              review={review.body}
              submit={submit}
              user={user}
            />
          )}
          {reviewsList &&
          !reviewsLoading &&
          !reviewsError &&
          reviewsList.length > 0 ? (
            <ul className="content-page__reviews-list">
              {reviewsList.map(rev => (
                <li className="content-page__reviews-item" key={rev.id}>
                  <Review
                    date={rev.created}
                    onRate={val => handleRating(val, content.id, true)}
                    onDelete={() => handleDeleteReview(rev.id)}
                    rating={rev.rating}
                    myRating={content.myRating}
                    review={rev.body}
                    user={user}
                    username={rev.author.username}
                  />
                </li>
              ))}
            </ul>
          ) : (
            !user && (
              <p>
                <Link className="cta content-page__link" to={routes.login.path}>
                  Login
                </Link>
                &nbsp; or &nbsp;
                <Link
                  className="cta content-page__link"
                  to={routes.register.path}
                >
                  register
                </Link>
                &nbsp; to create reviews
              </p>
            )
          )}
        </div>
      </section>
      {/* <section>
        <h2 className="content-page__header">Related</h2>
      </section> */}
    </div>
  );
};
