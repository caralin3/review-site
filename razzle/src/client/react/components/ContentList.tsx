import React from 'react';
import { ContentType, MultipleContentResponse, User } from '../../../common';
import {
  Banner,
  Container,
  ContentPreviewList,
  Tab,
  TabList,
  TabPanel
} from '../components';

export interface ContentListProps {
  content: MultipleContentResponse;
  handleRating: (value: number, id: string, rated: boolean) => void;
  handleWatch: (watching: boolean, id: string) => void;
  type: ContentType;
  user?: User;
}

export const ContentList: React.FC<ContentListProps> = ({
  content,
  handleRating,
  handleWatch,
  type,
  user
}) => {
  const [tab, setTab] = React.useState<'movie' | 'show' | 'watch'>('movie');
  const watchTab = React.useRef(null);
  const movieTab = React.useRef(null);
  const showTab = React.useRef(null);

  React.useEffect(() => {
    if (user) {
      setTab('watch');
    } else {
      if (type === 'Series') {
        setTab('show');
      }
    }
  }, [type]);

  const handleKeyDown = (e: React.KeyboardEvent<HTMLButtonElement>) => {
    if (e.keyCode === 37 || e.keyCode === 39) {
      if (document.activeElement && document.activeElement.id === 'watch') {
        if (type === 'Movie') {
          if (movieTab && movieTab.current) {
            (movieTab.current as any).focus();
          }
        } else {
          if (showTab && showTab.current) {
            (showTab.current as any).focus();
          }
        }
      } else if (
        document.activeElement &&
        document.activeElement.id === 'movie'
      ) {
        if (watchTab && watchTab.current) {
          (watchTab.current as any).focus();
        }
      } else if (
        document.activeElement &&
        document.activeElement.id === 'show'
      ) {
        if (watchTab && watchTab.current) {
          (watchTab.current as any).focus();
        }
      }
    }
  };

  return (
    <>
      <Banner>
        <i className="fas fa-film banner__icon" />
        <h1 className="banner__brand">Featured</h1>
      </Banner>
      <Container>
        <TabList>
          {user && (
            <Tab
              id="watch"
              innerRef={watchTab}
              panelId="watch-list"
              selected={tab}
              onClick={() => setTab('watch')}
              onKeyDown={handleKeyDown}
            >
              Watch List
            </Tab>
          )}
          {type === 'Movie' && (
            <Tab
              id="movie"
              innerRef={movieTab}
              panelId="movie-panel"
              selected={tab}
              onClick={() => setTab('movie')}
              onKeyDown={handleKeyDown}
            >
              Movies
            </Tab>
          )}
          {type === 'Series' && (
            <Tab
              id="show"
              innerRef={showTab}
              panelId="show-panel"
              selected={tab}
              onClick={() => setTab('show')}
              onKeyDown={handleKeyDown}
            >
              TV Shows
            </Tab>
          )}
        </TabList>
        {user && (
          <TabPanel selected={tab} id="watch-list" tabId="watch">
            <ContentPreviewList
              contentList={content.allContent.filter(
                con => con.watchList && con.type === type
              )}
              handleRating={handleRating}
              handleWatch={handleWatch}
              user={user}
            />
          </TabPanel>
        )}
        {type === 'Movie' && (
          <TabPanel selected={tab} id="movie-panel" tabId="movie">
            <ContentPreviewList
              contentList={content.allContent.filter(
                con => con.type === 'Movie'
              )}
              handleRating={handleRating}
              handleWatch={handleWatch}
              user={user}
            />
          </TabPanel>
        )}
        {type === 'Series' && (
          <TabPanel selected={tab} id="show-panel" tabId="show">
            <ContentPreviewList
              contentList={content.allContent.filter(
                con => con.type === 'Series'
              )}
              handleRating={handleRating}
              handleWatch={handleWatch}
              user={user}
            />
          </TabPanel>
        )}
      </Container>
    </>
  );
};
