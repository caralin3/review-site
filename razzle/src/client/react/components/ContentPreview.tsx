import React from 'react';
import { Link } from 'react-router-dom';
import { Content, User } from '../../../common';
import { routes } from '../routes';
import { StarRating, WatchButton } from '.';

export interface ContentPreviewProps {
  content: Content;
  onRate: (star: number) => void;
  onWatch: () => void;
  user?: User;
}

export const ContentPreview: React.FC<ContentPreviewProps> = ({
  content,
  onRate,
  onWatch,
  user
}) => {
  const titleSection = (
    <div className="content__header">
      <span className="content__title">
        <div>{content.title}</div>
        {content.network && (
          <span className="content__network">{content.network}</span>
        )}
      </span>
      <StarRating
        rating={content.rating}
        myRating={content.myRating}
        onClick={onRate}
        user={user}
      />
    </div>
  );

  const detailsSection = (
    <div className="content__details">
      <p>{content.type}</p>
      <p>
        {content.type === 'Movie'
          ? content.year
          : `${content.year}-${content.endYear ? content.endYear : ''}`}
      </p>
      <p>{content.mpa}</p>
      <p>{content.duration} mins</p>
      <p className="show-medium">{content.genres.toString()}</p>
      <p className="show-small">{content.genres.slice(0, 2).toString()}</p>
    </div>
  );

  return (
    <div className="content">
      <div className="show-small">
        {titleSection}
        {detailsSection}
      </div>
      <div className="content__mobile show-small">
        <img
          className="content__image"
          src={content.image}
          alt="poster image"
        />
        <p className="content__synopsis">
          {content.synopsis.substring(0, 175)}
          {content.synopsis.length > 175 && '...'}
          <span className="content__footer-mobile">
            <Link
              className="cta"
              to={
                content.type === 'Movie'
                  ? `${routes.movie.path}/${content.id}`
                  : `${routes.show.path}/${content.id}`
              }
            >
              Read more...
            </Link>
            {user && (
              <WatchButton watching={content.watchList} onClick={onWatch} />
            )}
          </span>
        </p>
      </div>
      <img
        className="content__image show-medium"
        src={content.image}
        alt="poster image"
      />
      <div className="content__text show-medium">
        <div>
          {titleSection}
          {detailsSection}
        </div>
        <p className="content__synopsis">
          {content.synopsis.substring(0, 450)}
          {content.synopsis.length > 450 && '...'}
        </p>
        <p className="content__footer">
          <Link
            className="cta"
            to={
              content.type === 'Movie'
                ? `${routes.movie.path}/${content.id}`
                : `${routes.show.path}/${content.id}`
            }
          >
            Read more...
          </Link>
          {user && (
            <WatchButton watching={content.watchList} onClick={onWatch} />
          )}
        </p>
      </div>
    </div>
  );
};
