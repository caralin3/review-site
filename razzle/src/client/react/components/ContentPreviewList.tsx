import React from 'react';
import { Content, User } from '../../../common';
import { ContentPreview } from '.';

export interface ContentPreviewListProps {
  contentList: Content[];
  handleRating: (value: number, id: string) => void;
  handleWatch: (watching: boolean) => void;
  user?: User;
}

export const ContentPreviewList: React.FC<ContentPreviewListProps> = ({
  contentList,
  handleRating,
  handleWatch,
  user
}) => (
  <ul className="content__list">
    {contentList.map(content => (
      <li className="content__item" key={content.id}>
        <ContentPreview
          content={content}
          onRate={val => handleRating(val, content.id)}
          onWatch={() => handleWatch(content.watchList)}
          user={user}
        />
      </li>
    ))}
  </ul>
);
