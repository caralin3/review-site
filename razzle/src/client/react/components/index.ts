export * from './Button';
export * from './Home';
