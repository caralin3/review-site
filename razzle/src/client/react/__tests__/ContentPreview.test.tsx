import React from 'react';
import { MemoryRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';
import { ContentPreview } from '../components';
import { Content } from '../../../common';
import { movie1, show1 } from '../../mock';

const onRate = jest.fn();
const onWatch = jest.fn();

const component = (content: Content) => {
  return (
    <MemoryRouter>
      <ContentPreview content={content} onRate={onRate} onWatch={onWatch}>
        <div>Content</div>
      </ContentPreview>
    </MemoryRouter>
  );
};

describe('ContentPreview', () => {
  it('renders movie content correctly', () => {
    const output = renderer.create(component(movie1)).toJSON();

    expect(output).toMatchSnapshot();
  });

  it('renders series content correctly', () => {
    const output = renderer.create(component(show1)).toJSON();

    expect(output).toMatchSnapshot();
  });
});
