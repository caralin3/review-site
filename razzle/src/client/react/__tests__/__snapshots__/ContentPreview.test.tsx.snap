// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ContentPreview renders movie content correctly 1`] = `
<div
  className="content-preview"
>
  <div
    className="show-small"
  >
    <div
      className="content-preview__header"
    >
      <span
        className="content-preview__title"
      >
        <div>
          Harry Potter and the Chamber of Secrets
        </div>
      </span>
      <div
        className="stars"
      >
        <i
          className="fas fa-star stars--active"
          onClick={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
        />
        <i
          className="fas fa-star stars--active"
          onClick={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
        />
        <i
          className="fas fa-star stars--active"
          onClick={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
        />
        <i
          className="fas fa-star stars--active"
          onClick={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
        />
        <i
          className="fas fa-star"
          onClick={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
        />
      </div>
    </div>
    <div
      className="content-preview__details"
    >
      <p>
        Movie
      </p>
      <p>
        2002
      </p>
      <p>
        PG
      </p>
      <p>
        95
         mins
      </p>
      <p
        className="show-medium"
      >
        Family
      </p>
      <p
        className="show-small"
      >
        Family
      </p>
    </div>
  </div>
  <div
    className="content-preview__mobile show-small"
  >
    <img
      alt="poster image"
      className="content-preview__image"
      src=""
    />
    <p
      className="content-preview__synopsis"
    >
      As students are paralyzed, Harry and the gang must go to the chamber  of secrets to find the culprit.
      <span
        className="content-preview__footer-mobile"
      >
        Read more...
        <button
          className="btn btn-sm btn-primary"
          onClick={[MockFunction]}
        >
          <span
            className="show-small"
          >
            <span>
              <i
                className="fas fa-plus"
              />
                Watch
               
            </span>
          </span>
          <span
            className="show-medium"
          >
            <span>
              <i
                className="fas fa-plus"
              />
                Add to Watch List
               
            </span>
          </span>
        </button>
      </span>
    </p>
  </div>
  <img
    alt="poster image"
    className="content-preview__image show-medium"
    src=""
  />
  <div
    className="content-preview__text show-medium"
  >
    <div>
      <div
        className="content-preview__header"
      >
        <span
          className="content-preview__title"
        >
          <div>
            Harry Potter and the Chamber of Secrets
          </div>
        </span>
        <div
          className="stars"
        >
          <i
            className="fas fa-star stars--active"
            onClick={[Function]}
            onMouseEnter={[Function]}
            onMouseLeave={[Function]}
          />
          <i
            className="fas fa-star stars--active"
            onClick={[Function]}
            onMouseEnter={[Function]}
            onMouseLeave={[Function]}
          />
          <i
            className="fas fa-star stars--active"
            onClick={[Function]}
            onMouseEnter={[Function]}
            onMouseLeave={[Function]}
          />
          <i
            className="fas fa-star stars--active"
            onClick={[Function]}
            onMouseEnter={[Function]}
            onMouseLeave={[Function]}
          />
          <i
            className="fas fa-star"
            onClick={[Function]}
            onMouseEnter={[Function]}
            onMouseLeave={[Function]}
          />
        </div>
      </div>
      <div
        className="content-preview__details"
      >
        <p>
          Movie
        </p>
        <p>
          2002
        </p>
        <p>
          PG
        </p>
        <p>
          95
           mins
        </p>
        <p
          className="show-medium"
        >
          Family
        </p>
        <p
          className="show-small"
        >
          Family
        </p>
      </div>
    </div>
    <p
      className="content-preview__synopsis"
    >
      As students are paralyzed, Harry and the gang must go to the chamber  of secrets to find the culprit.
    </p>
    <p
      className="content-preview__footer"
    >
      <a
        className="link"
        href="/movies/1fb69fe3-ac0b-49ce-9daf-f3e774eb981f"
        onClick={[Function]}
      >
        Read more...
      </a>
      <button
        className="btn btn-sm btn-primary"
        onClick={[MockFunction]}
      >
        <span
          className="show-small"
        >
          <span>
            <i
              className="fas fa-plus"
            />
              Watch
             
          </span>
        </span>
        <span
          className="show-medium"
        >
          <span>
            <i
              className="fas fa-plus"
            />
              Add to Watch List
             
          </span>
        </span>
      </button>
    </p>
  </div>
</div>
`;

exports[`ContentPreview renders series content correctly 1`] = `
<div
  className="content-preview"
>
  <div
    className="show-small"
  >
    <div
      className="content-preview__header"
    >
      <span
        className="content-preview__title"
      >
        <div>
          Younger
        </div>
        <span
          className="content-preview__network"
        >
          TV Land
        </span>
      </span>
      <div
        className="stars"
      >
        <i
          className="fas fa-star stars--fill"
          onClick={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
        />
        <i
          className="fas fa-star stars--fill"
          onClick={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
        />
        <i
          className="fas fa-star stars--fill"
          onClick={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
        />
        <i
          className="fas fa-star"
          onClick={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
        />
        <i
          className="fas fa-star"
          onClick={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
        />
      </div>
    </div>
    <div
      className="content-preview__details"
    >
      <p>
        Series
      </p>
      <p>
        2015-
      </p>
      <p>
        TV-14
      </p>
      <p>
        24
         mins
      </p>
      <p
        className="show-medium"
      >
        Comedy,Drama,Romance
      </p>
      <p
        className="show-small"
      >
        Comedy,Drama
      </p>
    </div>
  </div>
  <div
    className="content-preview__mobile show-small"
  >
    <img
      alt="poster image"
      className="content-preview__image"
      src=""
    />
    <p
      className="content-preview__synopsis"
    >
      A 40 year old pretends to be 26 to get back into the publishing world.    Lorem ipsum dolor sit amet, consectetur adipiscing elit,     sed do eiusmod tempor incididunt ut labo
      ...
      <span
        className="content-preview__footer-mobile"
      >
        Read more...
        <button
          className="btn btn-sm btn-primary btn-primary-fill"
          onClick={[MockFunction]}
        >
          <span
            className="show-small"
          >
            <span>
              <i
                className="fas fa-minus"
              />
                Unwatch
               
            </span>
          </span>
          <span
            className="show-medium"
          >
            <span>
              <i
                className="fas fa-minus"
              />
                Remove from Watch List
               
            </span>
          </span>
        </button>
      </span>
    </p>
  </div>
  <img
    alt="poster image"
    className="content-preview__image show-medium"
    src=""
  />
  <div
    className="content-preview__text show-medium"
  >
    <div>
      <div
        className="content-preview__header"
      >
        <span
          className="content-preview__title"
        >
          <div>
            Younger
          </div>
          <span
            className="content-preview__network"
          >
            TV Land
          </span>
        </span>
        <div
          className="stars"
        >
          <i
            className="fas fa-star stars--fill"
            onClick={[Function]}
            onMouseEnter={[Function]}
            onMouseLeave={[Function]}
          />
          <i
            className="fas fa-star stars--fill"
            onClick={[Function]}
            onMouseEnter={[Function]}
            onMouseLeave={[Function]}
          />
          <i
            className="fas fa-star stars--fill"
            onClick={[Function]}
            onMouseEnter={[Function]}
            onMouseLeave={[Function]}
          />
          <i
            className="fas fa-star"
            onClick={[Function]}
            onMouseEnter={[Function]}
            onMouseLeave={[Function]}
          />
          <i
            className="fas fa-star"
            onClick={[Function]}
            onMouseEnter={[Function]}
            onMouseLeave={[Function]}
          />
        </div>
      </div>
      <div
        className="content-preview__details"
      >
        <p>
          Series
        </p>
        <p>
          2015-
        </p>
        <p>
          TV-14
        </p>
        <p>
          24
           mins
        </p>
        <p
          className="show-medium"
        >
          Comedy,Drama,Romance
        </p>
        <p
          className="show-small"
        >
          Comedy,Drama
        </p>
      </div>
    </div>
    <p
      className="content-preview__synopsis"
    >
      A 40 year old pretends to be 26 to get back into the publishing world.    Lorem ipsum dolor sit amet, consectetur adipiscing elit,     sed do eiusmod tempor incididunt ut labore et dolore magna aliqua.     Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi     ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit     in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur     sint occaec
      ...
    </p>
    <p
      className="content-preview__footer"
    >
      <a
        className="link"
        href="/shows/444f885d-f4b5-4a06-8c5b-ae0e449c1ccd"
        onClick={[Function]}
      >
        Read more...
      </a>
      <button
        className="btn btn-sm btn-primary btn-primary-fill"
        onClick={[MockFunction]}
      >
        <span
          className="show-small"
        >
          <span>
            <i
              className="fas fa-minus"
            />
              Unwatch
             
          </span>
        </span>
        <span
          className="show-medium"
        >
          <span>
            <i
              className="fas fa-minus"
            />
              Remove from Watch List
             
          </span>
        </span>
      </button>
    </p>
  </div>
</div>
`;
